---
title: "ngs_final_integrated"
author: "Shaleigh Smith"
date: "5/2/2019"
output: html_document
---


---

---



---


---

Shaleigh Smith
NGS Final

---

Cellranger Results:
- Unfiltered: Contains every barcode from fixed list of known-good barcode sequences. This includes background and cell associated barcodes.
- Filtered: Contains only detected cellular barcodes.

---

General Steps:
1) Initialize a Seurat object
2) Filter cells based on built-in metrics
3) Normalize the data
4) Select variable genes across cells
5) Removing confounding factors by regression
6) Linear dimensionality reduction
7) PCA-based clustering
8) t-SNE & UMAP visualization 
9) Identify differentially expressed genes

---

# Helpful Links:

https://github.com/satijalab/seurat/wiki/Assay 
https://github.com/igordot/seurat/blob/master/R/preprocessing.R
https://satijalab.org/seurat/v3.0/immune_alignment.html 
https://github.com/satijalab/seurat/wiki 
https://broadinstitute.github.io/2019_scWorkshop/index.html 
https://hemberg-lab.github.io/scRNA.seq.course/index.html
#https://satijalab.org/seurat/v3.0/pbmc3k_tutorial.html


---


```{r}

# Load libraries
library(tidyverse)
library(dplyr)
library(plyr)
library(Matrix)
library(devtools)
library(Seurat)
library(ggplot2)
library(circlize)
library(data.table)
#library(cowplot)
library(DESeq2)
library(MAST)

#detach("package:cowplot", unload=TRUE)

packageVersion("Seurat")

```


---


Preprocessing

---


```{r}

# Set working directory
setwd("/Users/sha/Desktop/NGS_Informatics/NGS_final_project/")

### Preprocessing 
# Import matrices 
# Create Seurat objects 
# Set minimum for cells (3) and features (200) to remove low quality cells 

# Uninfected
data_dir_un <- "./uninfected/filtered_feature_bc_matrix/"
list.files(data_dir_un) # Should show barcodes.tsv, genes.tsv, and matrix.mtx
exp_matrix_un <- Read10X(data.dir = data_dir_un)
un <- CreateSeuratObject(counts = exp_matrix_un, project = "uninfected",
                         min.cells = 3, min.features = 200)
un$stage <- "Uninfected"
un


# Latent
data_dir_lat <- "./latent/filtered_feature_bc_matrix/"
list.files(data_dir_lat) # Should show barcodes.tsv, genes.tsv, and matrix.mtx
exp_matrix_lat <- Read10X(data.dir = data_dir_lat)
lat = CreateSeuratObject(counts = exp_matrix_lat, project = "latent",
                         min.cells = 3, min.features = 200)
lat$stage <- "Latent"
lat


# Ly20
data_dir_ly20 <- "./ly20/filtered_feature_bc_matrix/"
list.files(data_dir_ly20) # Should show barcodes.tsv, genes.tsv, and matrix.mtx
exp_matrix_ly20 <- Read10X(data.dir = data_dir_ly20)
ly20 = CreateSeuratObject(counts = exp_matrix_ly20, project = "ly20",
                         min.cells = 3, min.features = 200)
ly20$stage <- "LY20"
ly20


# Uninfected
data_dir_ly48 <- "./ly48/filtered_feature_bc_matrix/"
list.files(data_dir_ly48) # Should show barcodes.tsv, genes.tsv, and matrix.mtx
exp_matrix_ly48 <- Read10X(data.dir = data_dir_ly48)
ly48 = CreateSeuratObject(counts = exp_matrix_ly48, project = "ly48",
                         min.cells = 3, min.features = 200)
ly48$stage <- "LY48"
ly48 

```

```{r}

# View examples
colnames(un)
rownames(un)
head(x = un@meta.data, 20)
un[["RNA"]]@data

?seurat

```


```{r}

### Quality Control & Filtering

# View summary of total expression by single cell
#summary(colSums(un))
#summary(colSums(lat))
#summary(colSums(ly20))
#summary(colSums(ly48))

### Note that the mean decreases from uninfected to latent to ly 
### This suggests that host gene expression is decreasing as viral expression increases 


# Calculate percentage of all counts belonging to a subset of the possible features for each cell
VlnPlot(object = un, features = c("nFeature_RNA", "nCount_RNA"), ncol = 2, 
        cols = c("#007BC3"),  pt.size = 0.25)
VlnPlot(object = lat, features = c("nFeature_RNA", "nCount_RNA"), ncol = 2, 
        cols = c("#007BC3"),  pt.size = 0.25)
VlnPlot(object = ly20, features = c("nFeature_RNA", "nCount_RNA"), ncol = 2, 
        cols = c("#007BC3"),  pt.size = 0.25)
VlnPlot(object = ly48, features = c("nFeature_RNA", "nCount_RNA"), ncol = 2, 
        cols = c("#007BC3"),  pt.size = 0.25)

# Visualize feature relationships: Count_RNA vs. Feature_RNA
scatter_un_1 <- FeatureScatter(object = un, feature1 = "nCount_RNA", feature2 = "nFeature_RNA",
                               col = "darkblue") + xlim(c(0, 85000)) + ylim(c(0, 8000))
scatter_lat_1 <- FeatureScatter(object = lat, feature1 = "nCount_RNA", feature2 = "nFeature_RNA",
                                col = "darkred") + xlim(c(0, 85000)) + ylim(c(0, 8000))
scatter_ly20_1 <- FeatureScatter(object = ly20, feature1 = "nCount_RNA", feature2 = "nFeature_RNA",
                                 col = "darkgoldenrod2") + xlim(c(0, 85000)) + ylim(c(0, 8000))
scatter_ly48_1 <- FeatureScatter(object = ly48, feature1 = "nCount_RNA", feature2 = "nFeature_RNA",
                                 col = "darkgreen") + xlim(c(0, 85000)) + ylim(c(0, 8000))
scatter_un_1
scatter_lat_1
scatter_ly20_1
scatter_ly48_1
#CombinePlots(plots = list(scatter_un_1, scatter_lat_1, scatter_ly20_1, scatter_ly48_1), ncol = 2)

### All appear to be a horizontal parabola (k * sqrt(x) relationship)
### All have a fairly high positive correlation (greater than 0.87 )

```


```{r}

View(un)

# Distribution of expression levels by Feature and Cell
hist(rowSums(as.data.frame(un[["RNA"]]@data)),
     breaks = 100, main = "Uninfected: Expression sum by Feature",
     xlab = "Sum Expression")
hist(colSums(as.data.frame(un[["RNA"]]@data)),
     breaks = 100, main = "Uninfected: Expression sum by Cell",
     xlab = "Sum Expression")

hist(rowSums(as.data.frame(lat[["RNA"]]@data)),
     breaks = 100, main = "Latent: Expression sum by Feature",
     xlab = "Sum Expression")
hist(colSums(as.data.frame(lat[["RNA"]]@data)),
     breaks = 100, main = "Latent: Expression sum by Cell",
     xlab = "Sum Expression")

hist(rowSums(as.data.frame(ly20[["RNA"]]@data)),
     breaks = 100, main = "LY20: Expression sum by Feature",
     xlab = "Sum Expression")
hist(colSums(as.data.frame(ly20[["RNA"]]@data)),
     breaks = 100, main = "LY20: Expression sum by Cell",
     xlab = "Sum Expression")

hist(rowSums(as.data.frame(ly48[["RNA"]]@data)),
     breaks = 100, main = "LY48: Expression sum by Feature",
     xlab = "Sum Expression")
hist(colSums(as.data.frame(ly48[["RNA"]]@data)),
     breaks = 100, main = "LY48: Expression sum by Cell",
     xlab = "Sum Expression")

```


```{r}

### Remove unwanted cells from the dataset to decrease noise after normalization

# The normalized uninfected appears to be bimodal
# To decrease the smalled leftward peak, increase the minimum number of features by cell

un <- subset(x = un, subset = nFeature_RNA > 300 & nFeature_RNA < 6000)
lat <- subset(x = lat, subset = nFeature_RNA > 300 & nFeature_RNA < 6000)
ly20 <- subset(x = ly20, subset = nFeature_RNA > 300 & nFeature_RNA < 6000)
ly48 <- subset(x = ly48, subset = nFeature_RNA > 300 & nFeature_RNA < 6000)


```

```{r}

### Do normalization on subsetted matrix
# Use Log Normalization
# Alter subset to increase normal distribution

### Normalize each gene expression matrix 
# Do Log Normalization first with default scale

# Uninfected Log Normalization
un <- NormalizeData(un, assay = NULL, normalization.method = "LogNormalize", 
                         scale.factor = 10000)

hist(rowSums(as.data.frame(un[["RNA"]]@data)),
     breaks = 100, main = "Uninfected Sub Log Normalized: Expression sum by Feature",
     xlab = "Sum Expression")
hist(colSums(as.data.frame(un[["RNA"]]@data)),
     breaks = 100, main = "Uninfected Sub Log Normalized: Expression sum by Cell",
     xlab = "Sum Expression")

# Latent Log Normalization
lat <- NormalizeData(lat, assay = NULL, normalization.method = "LogNormalize", 
                         scale.factor = 10000)

hist(rowSums(as.data.frame(lat[["RNA"]]@data)),
     breaks = 100, main = "Latent Sub Log Normalized: Expression sum by Feature",
     xlab = "Sum Expression")
hist(colSums(as.data.frame(lat[["RNA"]]@data)),
     breaks = 100, main = "Latent Sub Log Normalized: Expression sum by Cell",
     xlab = "Sum Expression")

# ly20 Log Normalization
ly20 <- NormalizeData(ly20, assay = NULL, normalization.method = "LogNormalize", 
                         scale.factor = 10000)

hist(rowSums(as.data.frame(ly20[["RNA"]]@data)),
     breaks = 100, main = "LY20 Sub Log Normalized: Expression sum by Feature",
     xlab = "Sum Expression")
hist(colSums(as.data.frame(ly20[["RNA"]]@data)),
     breaks = 100, main = "LY20 Sub Log Normalized: Expression sum by Cell",
     xlab = "Sum Expression")

# ly48 Log Normalization
ly48 <- NormalizeData(ly48, assay = NULL, normalization.method = "LogNormalize", 
                         scale.factor = 10000)

hist(rowSums(as.data.frame(ly48[["RNA"]]@data)),
     breaks = 100, main = "LY48 Sub Log Normalized: Expression sum by Feature",
     xlab = "Sum Expression")
hist(colSums(as.data.frame(ly48[["RNA"]]@data)),
     breaks = 100, main = "LY48 Sub Log Normalized: Expression sum by Cell",
     xlab = "Sum Expression")

```

---

```{r}

### Feature Selection
# Identify the most highly variable features

un <- FindVariableFeatures(object = un, selection.method = "vst", nfeatures = 2000)
lat <- FindVariableFeatures(object = lat, selection.method = "vst", nfeatures = 2000)
ly20 <- FindVariableFeatures(object = ly20, selection.method = "vst", nfeatures = 2000)
ly48 <- FindVariableFeatures(object = ly48, selection.method = "vst", nfeatures = 2000)


# Identify the 10 most highly variable genes
un_top10 <- head(x = VariableFeatures(object = un), 10)
lat_top10 <- head(x = VariableFeatures(object = lat), 10)
ly20_top10 <- head(x = VariableFeatures(object = ly20), 10)
ly48_top10 <- head(x = VariableFeatures(object = ly48), 10)

# Plot variable features with labels
un_feat_plot <- LabelPoints(plot = VariableFeaturePlot(object = un, 
                                                       cols = c("black", "darkblue")), 
                            points = un_top10, repel = TRUE)
lat_feat_plot <- LabelPoints(plot = VariableFeaturePlot(object = lat, 
                                                        cols = c("black", "darkred")), 
                            points = lat_top10, repel = TRUE)
ly20_feat_plot <- LabelPoints(plot = VariableFeaturePlot(object = ly20, 
                                                         cols = c("black", "darkgoldenrod2")), 
                            points = ly20_top10, repel = TRUE)
ly48_feat_plot <- LabelPoints(plot = VariableFeaturePlot(object = ly48, 
                                                         cols = c("black", "darkgreen")), 
                            points = ly48_top10, repel = TRUE)

un_feat_plot
lat_feat_plot
ly20_feat_plot
ly48_feat_plot

```


```{r}

# Integrate the four datasets 

# Identify anchors to integrate the 4 datasets 
stage_anchors <- FindIntegrationAnchors(object.list = list(un, lat, ly20, ly48), dims = 1:50)
stage_combined <- IntegrateData(anchorset = stage_anchors, dims = 1:50)

# Label assay
DefaultAssay(object = stage_combined) <- "integrated"

```


```{r}

# Run the standard workflow for visualization and clustering
stage_combined <- ScaleData(object = stage_combined, verbose = FALSE)
stage_combined <- RunPCA(object = stage_combined, npcs = 50, verbose = FALSE)

### Determining the Dimensionality of the dataset
# How many PCs are optimal 

# Elbow Plot
ElbowPlot(object = stage_combined) # ?????? PCs

# Compare the distribution of p-values for each PC with a uniform distribution (dashed line)
#stage_combined <- JackStraw(object = stage_combined, num.replicate = 100)
#stage_combined <- ScoreJackStraw(object = stage_combined, dims = 1:20)
#JackStrawPlot(object = stage_combined, dims = 1:20)


```

```{r}

set.seed(1234)

# t-SNE and Clustering
stage_combined <- RunTSNE(object = stage_combined, reduction = "pca", dims = 1:10) # was 20
stage_combined <- RunUMAP(object = stage_combined, reduction = "pca", dims = 1:10) # was 20
stage_combined <- FindNeighbors(object = stage_combined, reduction = "pca", dims = 1:10) # was 20
stage_combined <- FindClusters(stage_combined, resolution = 0.2) # was 0.5 (decreasing this decreases number of clusters)

head(x = Idents(object = stage_combined), 10)

```


```{r}

# Save object
#saveRDS(stage_combined, file = "./stage_integrated_may_5.rds")

```


```{r}

# Read in object

stage_combined <- readRDS(file = "./stage_integrated.rds")

#head(Idents(stage_combined))

```


```{r}

# Visualizations

# PCA
pca_1 <- DimPlot(object = stage_combined, reduction = "pca", group.by = "stage")
pca_2 <- DimPlot(object = stage_combined, reduction = "pca", label = TRUE)
pca_3 <- DimPlot(object = stage_combined, reduction = "pca", split.by = "stage")
pca_1
pca_2
pca_3

# TSNE
tsne_1 <- DimPlot(object = stage_combined, reduction = "tsne", group.by = "stage")
tsne_2 <- DimPlot(object = stage_combined, reduction = "tsne", label = TRUE)
tsne_3 <- DimPlot(object = stage_combined, reduction = "tsne", split.by = "stage")
tsne_1
tsne_2
tsne_3

# UMAP
umap_1 <- DimPlot(object = stage_combined, reduction = "umap", group.by = "stage")
umap_2 <- DimPlot(object = stage_combined, reduction = "umap", label = TRUE)
umap_3 <- DimPlot(object = stage_combined, reduction = "umap", split.by = "stage")
umap_1
umap_2
umap_3

```

----


https://learn.gencore.bio.nyu.edu/single-cell-rnaseq/seurat-part-4-cell-clustering/

Compare genes in groups that are visually by each other in tsne 




----


```{r}

# Find markers for each cluster

# Only positive markers 
clust_markers <- FindAllMarkers(object = stage_combined, only.pos = TRUE)
clust_markers_10 <- clust_markers %>% group_by(cluster) %>% top_n(n = 10, wt = avg_logFC)
clust_markers_10

write.table(clust_markers_10, file = "top10_cluster_markers.tsv", sep = "\t", quote = FALSE)

```


```{r}

### TEst top 10
#clust_markers_20_merge <- clust_markers_20

#clust_markers_20_merge$gene <- toupper(clust_markers_20_merge$gene)
#clust_markers_20_merge

# Read in minimal cell classifier data frame 
#cell_class_min <- read_csv("cell_classifier_minimal.csv", col_names = FALSE)
#colnames(cell_class_min)[1:2] <- c("cell_type", "gene")
#cell_class_min <- dplyr::select(cell_class_min, cell_type, gene)

# Intersect cluster dataframes with cell classifier minimal dataframe 
#cell_class_cluster_min <- inner_join(clust_markers_20_merge, cell_class, by = "gene")
#cell_class_cluster_min

```


```{r}

# Find conserved cell markers 
DefaultAssay(object = stage_combined) <- "RNA"

markers_9 <- FindConservedMarkers(object = stage_combined, ident.1 = 9,
                                  grouping.var = "stage", 
                                  verbose = FALSE)
head(x = markers_9, n = 10)
markers_9

markers_3 <- FindConservedMarkers(object = stage_combined, ident.1 = 3,
                                  grouping.var = "stage", 
                                  verbose = FALSE)
head(x = markers_3, n = 10)
markers_3


markers_7 <- FindConservedMarkers(object = stage_combined, ident.1 = 7,
                                  grouping.var = "stage", 
                                  verbose = FALSE)
head(x = markers_7, n = 10)
markers_7


markers_3_7 <- FindConservedMarkers(object = stage_combined, ident.1 = 3, ident.2 = 7,
                                  grouping.var = "stage", 
                                  verbose = FALSE)
head(x = markers_3_7, n = 10)
markers_3_7

# 6 & 5 aren't found in Latent #
markers_0_2_4_5 <- FindConservedMarkers(object = stage_combined, ident.1 = c(0,2,4,5),
                                       ident.2 = c(1,3,7,8,9),
                                  grouping.var = "stage", 
                                  verbose = FALSE)
head(x = markers_0_2_4_5, n = 10)
markers_0_2_4_5

markers_1_4_5 <- FindConservedMarkers(object = stage_combined, ident.1 = c(1,4,5),
                                  grouping.var = "stage", 
                                  verbose = FALSE)
head(x = markers_1_4_5, n = 10)
markers_1_4_5


markers_2_8 <- FindConservedMarkers(object = stage_combined, ident.1 = c(2,8),
                                  grouping.var = "stage", 
                                  verbose = FALSE)
head(x = markers_2_8, n = 10)
markers_2_8


### Other (because cluster 5/6 isn't found in latent)
stage_combined_sub_lat <- SubsetData(object = stage_combined, 
                                 subset.name = "stage", 
                                accept.value = c("Uninfected", "LY20", "LY48"))
markers_0_6 <- FindConservedMarkers(object = stage_combined_sub_lat, ident.1 = c(0,6),
                                    ident.2 = c(1,2,3,4,5,7,8,9),
                                    grouping.var = "stage",
                                    verbose = FALSE)
head(x = markers_0_6, n = 10)
markers_0_6

markers_5 <- FindConservedMarkers(object = stage_combined_sub_lat, ident.1 = 5,
                                  grouping.var = "stage", 
                                  verbose = FALSE)
head(x = markers_5, n = 10)
markers_5



```


---

```{r}

# Re-categorize these

markers_macrophage <- markers_3
markers_schwann <- markers_7
markers_cell_stress <- markers_9
#markers_3_7
markers_neuron_npy_pos <- markers_0_6
markers_neuron_npy_neg <- markers_1_4_5
markers_satellite <- markers_2_8


# View Data frames
markers_macrophage
markers_schwann
markers_cell_stress
markers_neuron_npy_pos 
markers_neuron_npy_neg
markers_satellite

# Write out these tables 
#write.table(markers_macrophage, file = "markers_macrophage.tsv", sep = "\t", quote = FALSE)
#write.table(markers_schwann, file = "markers_schwann.tsv", sep = "\t", quote = FALSE)
#write.table(markers_cell_stress, file = "markers_cell_stress.tsv", sep = "\t", quote = FALSE)
#write.table(markers_neuron_npy_pos, file = "markers_neuron_npy_pos.tsv", sep = "\t", quote = FALSE)
#write.table(markers_neuron_npy_neg, file = "markers_neuron_npy_neg.tsv", sep = "\t", quote = FALSE)
#write.table(markers_satellite, file = "markers_satellite.tsv", sep = "\t", quote = FALSE)


macro_temp <- read.table("./markers_macrophage.tsv", sep = "\t")
macro_temp

```


---

Clustering (dims = 1:10, resolution = 0.2)

NPY is more associated with Noradrenergic Neurons (but not 100%)

NPY+ are vasomotor neurons (Vasoconstrictor)
NPY- are pilomotor, secremotor, iridomotor, and thermogenic (Non-Vasoconstrictor)

VIP/RET/VACHT are Cholinergic Phenotype 


•	Cluster 0 - NPY+ Neuron
         -	TUBB2A
         -	TUBB2b
         -	TUBB3
         -	RESP18
         -  NPY
•	Cluster 1 - NPY- Neuron
         -	GNG2
         -	PCP4
         -  RET
•	Cluster 2 - Satellite
         -	SPARC
•	Cluster 3 – Macrophage 
         -	CD68 
         -	CD53
         -	PF4
         -	AIF1
         -	LYZ2
         -	GPNMB
•	Cluster 4 - NPY- Neuron 
         -	RET
         -  GNG2
•	Cluster 5 – NPY- Neuron 
         -	Prph
         -  RET
•	Cluster 6 – NPY+ Neuron 
         -	RESP18
         -	VGF
         -	NPY
•	Cluster 7 – Schwann Cells 
         -	S100B
         -	S0X10
•	Cluster 8 - Satellite
         -	SPARC
         -  Igfbp7 
         -  Col3a1
•	Cluster 9 – Cell Stress
         -	GADD45A
         -	DDIT3
         -	BBC3
         -	PPP1R15A
         -	CDKN1A

---

```{r}

# Visualize Markers

# Neurons
FeaturePlot(object = stage_combined, 
            features = c("Tubb3", "Gng2", "Prph","Vgf","Npy"),
            min.cutoff = "q9")

FeaturePlot(object = stage_combined, 
            features = c("Serp2", "Ccdc92", "Ctxn1", "Atp6v1a"),
            min.cutoff = "q9")

FeaturePlot(object = stage_combined, 
            features = c("Npy"),
            min.cutoff = "q9")


# Schwann Cells
FeaturePlot(object = stage_combined, 
            features = c("S100b", "Sox10"), 
            min.cutoff = "q9")

# Satellite Cells
FeaturePlot(object = stage_combined, 
            features = c("Sparc", "Igfbp7", "Col3a1"), 
            min.cutoff = "q9")

# Cell Stress is uniform but upregulated in cluster 9
# Lets remove high levels of Ddit3 and Ppp1r15a?
FeaturePlot(object = stage_combined, 
            features = c("Ddit3", "Ppp1r15a"), 
            min.cutoff = "q9")

# Macrophage 
FeaturePlot(object = stage_combined, 
            features = c("Cd68", "Cd53"), 
            min.cutoff = "q9")


## ALL
all_features <- FeaturePlot(object = stage_combined,
                            features = c("Tubb3", "Gng2", "Prph","Vgf","Npy",
                         "S100b", "Sox10",
                         "Sparc", "Igfbp7", "Col3a1",
                         "Ddit3", "Ppp1r15a",
                         "Cd68", "Cd53"),
                         min.cutoff = "q9",
                         label.size = 2,
                         combine = FALSE) 

for(i in 1:length(all_features)) {
  all_features[[i]] <- all_features[[i]] + NoLegend() + NoAxes()
}

cowplot::plot_grid(plotlist = all_features)

# Important Markers:

### Neurons
# Tubb3
# Gng2
# Prph
# Vgf
# Npy
# Ret

### Schwann Cells
# S100b
# Sox10
# Vim

### Satellite Cells
# Sparc
# Igfbp7
# Col3a1

### Macrophages
# Cd68
# Cd53

### Cell Stress Indicators 
# Ddit3
# Ppp1r15a

```


---

```{r}

# Remove cells under cell stress (cluster 9)
#stage_combined <- SubsetData(stage_combined, ident.remove = 9)

```


---

```{r}

# Create expression plots for cell markers 
VlnPlot(object = stage_combined, features = c("Tubb3", "Gng2", "Prph","Vgf","Npy"), pt.size = 0.01)
VlnPlot(object = stage_combined, features = c("S100b", "Sox10"), pt.size = 0.01)
VlnPlot(object = stage_combined, features = c("Sparc", "Igfbp7", "Col3a1"), pt.size = 0.01)
VlnPlot(object = stage_combined, features = c("Cd68", "Cd53"), pt.size = 0.01)
VlnPlot(object = stage_combined, features = c("Ddit3", "Ppp1r15a"), pt.size = 0.01)


```

```{r}

# Rename clusters

stage_combined <- RenameIdents(object = stage_combined, 
                               `0` = "NPY+ Neuron", 
                               `1` = "NPY- Neuron", 
                               `2` = "Satellite", 
                               `3` = "Macrophage", 
                               `4` = "NPY- Neuron", 
                               `5` = "NPY- Neuron", 
                               `6` = "NPY+ Neuron", 
                               `7` = "Schwann", 
                               `8` = "Satellite",
                               `9` = "Stress+ Neuron")

# Plots

tsne_cell_1 <- DimPlot(object = stage_combined, reduction = "tsne", label = TRUE)
tsne_cell_2 <- DimPlot(object = stage_combined, reduction = "tsne", split.by = "stage")

umap_cell_1 <- DimPlot(object = stage_combined, reduction = "umap", label = TRUE)
umap_cell_2 <- DimPlot(object = stage_combined, reduction = "umap", split.by = "stage")

tsne_cell_1
umap_cell_1
tsne_cell_2
umap_cell_2

```




---



Differential Expression



---

```{r}

# View conserved cell markers across conditions
plot_markers <- c("Tubb3", "Gng2", "Prph","Vgf","Npy",
                  "Serp2", "Ccdc92", "Ctxn1", "Ret",
                  "S100b", "Sox10",
                  "Sparc", "Igfbp7", "Col3a1",
                  "Cd68", "Cd53", 
                  "Ddit3", "Ppp1r15a")

DotPlot(object = stage_combined, features = plot_markers, 
        cols = c("darkblue", "darkred", "darkgoldenrod2", "darkgreen"),
        split.by = "stage", dot.min = 0.2) + RotatedAxis()

```


```{r}

# Identify differential expression across conditions

neuron_npy_pos_cells <- subset(x = stage_combined, idents = "NPY+ Neuron")
Idents(object = neuron_npy_pos_cells) <- "stage"
avg_neuron_npy_pos_cells <- log1p(x = AverageExpression(object = neuron_npy_pos_cells, verbose = FALSE)$RNA)
avg_neuron_npy_pos_cells$gene <- rownames(x = avg_neuron_npy_pos_cells)

neuron_npy_neg_cells <- subset(x = stage_combined, idents = "NPY- Neuron")
Idents(object = neuron_npy_neg_cells) <- "stage"
avg_neuron_npy_neg_cells <- log1p(x = AverageExpression(object = neuron_npy_neg_cells, verbose = FALSE)$RNA)
avg_neuron_npy_neg_cells$gene <- rownames(x = avg_neuron_npy_neg_cells)

satellite_cells <- subset(x = stage_combined, idents = "Satellite")
Idents(object = satellite_cells) <- "stage"
avg_satellite_cells <- log1p(x = AverageExpression(object = satellite_cells, verbose = FALSE)$RNA)
avg_satellite_cells$gene <- rownames(x = avg_satellite_cells)

schwann_cells <- subset(x = stage_combined, idents = "Schwann")
Idents(object = schwann_cells) <- "stage"
avg_schwann_cells <- log1p(x = AverageExpression(object = schwann_cells, verbose = FALSE)$RNA)
avg_schwann_cells$gene <- rownames(x = avg_schwann_cells)

macro_cells <- subset(x = stage_combined, idents = "Macrophage")
Idents(object = macro_cells) <- "stage"
avg_macro_cells <- log1p(x = AverageExpression(object = macro_cells, verbose = FALSE)$RNA)
avg_macro_cells$gene <- rownames(x = avg_macro_cells)

stress_cells <- subset(x = stage_combined, idents = "Stress+ Neuron")
Idents(object = stress_cells) <- "stage"
avg_stress_cells <- log1p(x = AverageExpression(object = stress_cells, verbose = FALSE)$RNA)
avg_stress_cells$gene <- rownames(x = avg_stress_cells)

# Plot Uninfected to Latent
diff_un_lat_neuron_npy_pos <- ggplot(avg_neuron_npy_pos_cells, aes(Uninfected, Latent)) + geom_point() + ggtitle("NPY+ Neurons")
diff_un_lat_neuron_npy_pos <- LabelPoints(plot = diff_un_lat_neuron_npy_pos, points = plot_markers, repel = TRUE)

diff_un_lat_neuron_npy_neg <- ggplot(avg_neuron_npy_neg_cells, aes(Uninfected, Latent)) + geom_point() + ggtitle("NPY- Neurons")
diff_un_lat_neuron_npy_neg <- LabelPoints(plot = diff_un_lat_neuron_npy_neg, points = plot_markers, repel = TRUE)

diff_un_lat_satellite <- ggplot(avg_satellite_cells, aes(Uninfected, Latent)) + geom_point() + ggtitle("Satellite Cells")
diff_un_lat_satellite <- LabelPoints(plot = diff_un_lat_satellite, points = plot_markers, repel = TRUE)

diff_un_lat_schwann <- ggplot(avg_schwann_cells, aes(Uninfected, Latent)) + geom_point() + ggtitle("Schwann Cells")
diff_un_lat_schwann <- LabelPoints(plot = diff_un_lat_schwann, points = plot_markers, repel = TRUE)

diff_un_lat_macro <- ggplot(avg_macro_cells, aes(Uninfected, Latent)) + geom_point() + ggtitle("Macrophages")
diff_un_lat_macro <- LabelPoints(plot = diff_un_lat_macro, points = plot_markers, repel = TRUE)

diff_un_lat_stress <- ggplot(avg_stress_cells, aes(Uninfected, Latent)) + geom_point() + ggtitle("Stress+ Neurons")
diff_un_lat_stress <- LabelPoints(plot = diff_un_lat_stress, points = plot_markers, repel = TRUE)

diff_un_lat_neuron_npy_pos
diff_un_lat_neuron_npy_neg
diff_un_lat_satellite
diff_un_lat_schwann
diff_un_lat_macro
diff_un_lat_stress
#plot_grid(diff_un_lat_neuron, diff_un_lat_satellite, diff_un_lat_schwann, diff_un_lat_macro)


# Plot Latent to LY20

test_mark <- c("Vim", "Trim2", "Scgb1c1", "Grm2", "Ldah", "Cd9", "Vip")

diff_lat_ly20_neuron_npy_pos <- ggplot(avg_neuron_npy_pos_cells, aes(Latent, LY20)) + geom_point() + ggtitle("NPY+ Neurons")
diff_lat_ly20_neuron_npy_pos <- LabelPoints(plot = diff_lat_ly20_neuron_npy_pos, points = test_mark, repel = TRUE)

diff_lat_ly20_neuron_npy_neg <- ggplot(avg_neuron_npy_neg_cells, aes(Latent, LY20)) + geom_point() + ggtitle("NPY- Neurons")
diff_lat_ly20_neuron_npy_neg <- LabelPoints(plot = diff_lat_ly20_neuron_npy_neg, points = plot_markers, repel = TRUE)

diff_lat_ly20_satellite <- ggplot(avg_satellite_cells, aes(Latent, LY20)) + geom_point() + ggtitle("Satellite Cells")
diff_lat_ly20_satellite <- LabelPoints(plot = diff_lat_ly20_satellite, points = plot_markers, repel = TRUE)

diff_lat_ly20_schwann <- ggplot(avg_schwann_cells, aes(Latent, LY20)) + geom_point() + ggtitle("Schwann Cells")
diff_lat_ly20_schwann <- LabelPoints(plot = diff_lat_ly20_schwann, points = plot_markers, repel = TRUE)

diff_lat_ly20_macro <- ggplot(avg_macro_cells, aes(Latent, LY20)) + geom_point() + ggtitle("Macrophages")
diff_lat_ly20_macro <- LabelPoints(plot = diff_lat_ly20_macro, points = plot_markers, repel = TRUE)

diff_lat_ly20_stress <- ggplot(avg_macro_cells, aes(Latent, LY20)) + geom_point() + ggtitle("Stress+ Neurons")
diff_lat_ly20_stress <- LabelPoints(plot = diff_lat_ly20_stress, points = plot_markers, repel = TRUE)

diff_lat_ly20_neuron_npy_pos
diff_lat_ly20_neuron_npy_neg
diff_lat_ly20_satellite
diff_lat_ly20_schwann
diff_lat_ly20_macro
diff_lat_ly20_stress
#plot_grid(diff_lat_ly20_neuron, diff_lat_ly20_satellite, diff_lat_ly20_schwann, diff_lat_ly20_macro)


# Plot LY20 to LY48
diff_ly20_ly48_neuron_npy_pos <- ggplot(avg_neuron_npy_pos_cells, aes(LY20, LY48)) + geom_point() + ggtitle("NPY+ Neurons")
diff_ly20_ly48_neuron_npy_pos <- LabelPoints(plot = diff_ly20_ly48_neuron_npy_pos, points = plot_markers, repel = TRUE)

diff_ly20_ly48_neuron_npy_neg <- ggplot(avg_neuron_npy_neg_cells, aes(LY20, LY48)) + geom_point() + ggtitle("NPY- Neurons")
diff_ly20_ly48_neuron_npy_neg <- LabelPoints(plot = diff_ly20_ly48_neuron_npy_neg, points = plot_markers, repel = TRUE)

diff_ly20_ly48_satellite <- ggplot(avg_satellite_cells, aes(LY20, LY48)) + geom_point() + ggtitle("Satellite Cells")
diff_ly20_ly48_satellite <- LabelPoints(plot = diff_ly20_ly48_satellite, points = plot_markers, repel = TRUE)

diff_ly20_ly48_schwann <- ggplot(avg_schwann_cells, aes(LY20, LY48)) + geom_point() + ggtitle("Schwann Cells")
diff_ly20_ly48_schwann <- LabelPoints(plot = diff_ly20_ly48_schwann, points = plot_markers, repel = TRUE)

diff_ly20_ly48_macro <- ggplot(avg_macro_cells, aes(LY20, LY48)) + geom_point() + ggtitle("Macrophages")
diff_ly20_ly48_macro <- LabelPoints(plot = diff_ly20_ly48_macro, points = plot_markers, repel = TRUE)

diff_ly20_ly48_stress <- ggplot(avg_macro_cells, aes(LY20, LY48)) + geom_point() + ggtitle("Stress+ Neurons")
diff_ly20_ly48_stress <- LabelPoints(plot = diff_ly20_ly48_stress, points = plot_markers, repel = TRUE)

diff_ly20_ly48_neuron_npy_pos
diff_ly20_ly48_neuron_npy_neg
diff_ly20_ly48_satellite
diff_ly20_ly48_schwann
diff_ly20_ly48_macro
diff_ly20_ly48_stress
#plot_grid(diff_ly20_ly48_neuron, diff_ly20_ly48_satellite, diff_ly20_ly48_schwann, diff_ly20_ly48_macro)


```




---

https://satijalab.org/seurat/v3.0/de_vignette.html 

Which Statistical test to use for identifying up and down regualted gene expression?

The following differential expression tests are currently supported:

“wilcox” : Wilcoxon rank sum test (default)
“bimod” : Likelihood-ratio test for single cell feature expression, (McDavid et al., Bioinformatics, 2013)
“roc” : Standard AUC classifier
“t” : Student’s t-test
“poisson” : Likelihood ratio test assuming an underlying negative binomial distribution. Use only for UMI-based datasets
“negbinom” : Likelihood ratio test assuming an underlying negative binomial distribution. Use only for UMI-based datasets
“LR” : Uses a logistic regression framework to determine differentially expressed genes. Constructs a logistic regression model predicting group membership based on each feature individually and compares this to a null model with a likelihood ratio test.
“MAST” : GLM-framework that treates cellular detection rate as a covariate (Finak et al, Genome Biology, 2015) (Installation instructions)
“DESeq2” : DE based on a model using the negative binomial distribution (Love et al, Genome Biology, 2014) (Installation instructions)



Notes:
- T-test assumes normal distribution
- Wilcox doesn't need normal distribution - alternative to t-test 
- ROC will calculate sensitivity and specificity 
- 


---


```{r}

stage_combined$celltype_stage <- paste(Idents(object = stage_combined), stage_combined$stage, sep = "_")
stage_combined$celltype <- Idents(object = stage_combined)
Idents(stage_combined) <- "celltype_stage"
Idents(stage_combined)

```




```{r}

# Pre-filter features that have less than a two-fold change between the average expression
# Neurons: Find differences between Latent and Uninfected
neuron_npy_pos_un_lat_markers <- FindMarkers(object = stage_combined, ident.1 = "NPY+ Neuron_Latent", 
                                             ident.2 = "NPY+ Neuron_Uninfected",
                                             verbose = FALSE, 
                                             #logfc.threshold = log(2), 
                                             test.use = "wilcox")
setDT(neuron_npy_pos_un_lat_markers, keep.rownames = TRUE)[]
colnames(neuron_npy_pos_un_lat_markers)[1] <- c("gene")
#neuron_npy_pos_un_lat_markers <- filter(neuron_npy_pos_un_lat_markers, p_val_adj <= 0.01)

# NPY- Neuron doesn't pass threshold
neuron_npy_neg_un_lat_markers <- FindMarkers(object = stage_combined, 
                                             ident.1 = "NPY- Neuron_Latent", 
                                             ident.2 = "NPY- Neuron_Uninfected",
                                             verbose = FALSE, 
                                             #logfc.threshold = log(2), 
                                             test.use = "wilcox")
setDT(neuron_npy_neg_un_lat_markers, keep.rownames = TRUE)[]
colnames(neuron_npy_neg_un_lat_markers)[1] <- c("gene")
#neuron_npy_neg_un_lat_markers <- filter(neuron_npy_neg_un_lat_markers, p_val_adj <= 0.01)

satellite_un_lat_markers <- FindMarkers(object = stage_combined, 
                                        ident.1 = "Satellite_Latent", 
                                        ident.2 = "Satellite_Uninfected",
                                        verbose = FALSE, 
                                        #logfc.threshold = log(2), 
                                        test.use = "wilcox")
setDT(satellite_un_lat_markers, keep.rownames = TRUE)[]
colnames(satellite_un_lat_markers)[1] <- c("gene")
#satellite_un_lat_markers <- filter(satellite_un_lat_markers, p_val_adj <= 0.01)

schwann_un_lat_markers <- FindMarkers(object = stage_combined, 
                                      ident.1 = "Schwann_Latent", 
                                      ident.2 = "Schwann_Uninfected",
                                      verbose = FALSE, 
                                      #logfc.threshold = log(2), 
                                      test.use = "wilcox")
setDT(schwann_un_lat_markers, keep.rownames = TRUE)[]
colnames(schwann_un_lat_markers)[1] <- c("gene")
#schwann_un_lat_markers <- filter(schwann_un_lat_markers, p_val_adj <= 0.01)

macro_un_lat_markers <- FindMarkers(object = stage_combined, 
                                    ident.1 = "Macrophage_Latent", 
                                    ident.2 = "Macrophage_Uninfected",
                                    verbose = FALSE, 
                                    #logfc.threshold = log(2), 
                                    test.use = "wilcox")
setDT(macro_un_lat_markers, keep.rownames = TRUE)[]
colnames(macro_un_lat_markers)[1] <- c("gene")
#macro_un_lat_markers <- filter(macro_un_lat_markers, p_val_adj <= 0.01)

stress_un_lat_markers <- FindMarkers(object = stage_combined, 
                                     ident.1 = "Stress+ Neuron_Latent", 
                                     ident.2 = "Stress+ Neuron_Uninfected",
                                     verbose = FALSE, 
                                     #logfc.threshold = log(2), 
                                     test.use = "wilcox")
setDT(stress_un_lat_markers, keep.rownames = TRUE)[]
colnames(stress_un_lat_markers)[1] <- c("gene")
#stress_un_lat_markers <- filter(stress_un_lat_markers, p_val_adj <= 0.01)


neuron_npy_pos_un_lat_markers #11
satellite_un_lat_markers #0
schwann_un_lat_markers #2
macro_un_lat_markers #0
stress_un_lat_markers #0

```

```{r}

# Pre-filter features that have less than a two-fold change between the average expression
# Neurons: Find differences between Ly20 and Latent
neuron_npy_pos_lat_ly20_markers <- FindMarkers(object = stage_combined, 
                                               ident.1 = "NPY+ Neuron_LY20", 
                                               ident.2 = "NPY+ Neuron_Latent",
                                               verbose = FALSE,
                                               #logfc.threshold = log(2),
                                               test.use = "wilcox")
setDT(neuron_npy_pos_lat_ly20_markers, keep.rownames = TRUE)[]
colnames(neuron_npy_pos_lat_ly20_markers)[1] <- c("gene")
#neuron_npy_pos_lat_ly20_markers <- filter(neuron_npy_pos_lat_ly20_markers, p_val_adj <= 0.01)

neuron_npy_neg_lat_ly20_markers <- FindMarkers(object = stage_combined, 
                                               ident.1 = "NPY- Neuron_LY20", 
                                               ident.2 = "NPY- Neuron_Latent",
                                               verbose = FALSE, 
                                               #logfc.threshold = log(2), 
                                               test.use = "wilcox")
setDT(neuron_npy_neg_lat_ly20_markers, keep.rownames = TRUE)[]
colnames(neuron_npy_neg_lat_ly20_markers)[1] <- c("gene")
#neuron_npy_neg_lat_ly20_markers <- filter(neuron_npy_neg_lat_ly20_markers, p_val_adj <= 0.01)

satellite_lat_ly20_markers <- FindMarkers(object = stage_combined, 
                                          ident.1 = "Satellite_LY20", 
                                          ident.2 = "Satellite_Latent",
                                          verbose = FALSE, 
                                          #logfc.threshold = log(2), 
                                          test.use = "wilcox")
setDT(satellite_lat_ly20_markers, keep.rownames = TRUE)[]
colnames(satellite_lat_ly20_markers)[1] <- c("gene")
#satellite_lat_ly20_markers <- filter(satellite_lat_ly20_markers, p_val_adj <= 0.01)

schwann_lat_ly20_markers <- FindMarkers(object = stage_combined, 
                                        ident.1 = "Schwann_LY20", 
                                        ident.2 = "Schwann_Latent", 
                                        verbose = FALSE, 
                                        #logfc.threshold = log(2), 
                                        test.use = "wilcox")
setDT(schwann_lat_ly20_markers, keep.rownames = TRUE)[]
colnames(schwann_lat_ly20_markers)[1] <- c("gene")
#schwann_lat_ly20_markers <- filter(schwann_lat_ly20_markers, p_val_adj <= 0.01)

macro_lat_ly20_markers <- FindMarkers(object = stage_combined, 
                                      ident.1 = "Macrophage_LY20", 
                                      ident.2 = "Macrophage_Latent",
                                      verbose = FALSE, 
                                      #logfc.threshold = log(2), 
                                      test.use = "wilcox")
setDT(macro_lat_ly20_markers, keep.rownames = TRUE)[]
colnames(macro_lat_ly20_markers)[1] <- c("gene")
#macro_lat_ly20_markers <- filter(macro_lat_ly20_markers, p_val_adj <= 0.01)

stress_lat_ly20_markers <- FindMarkers(object = stage_combined, 
                                       ident.1 = "Stress+ Neuron_LY20", 
                                       ident.2 = "Stress+ Neuron_Latent",
                                       verbose = FALSE, 
                                       #logfc.threshold = log(2), 
                                       test.use = "wilcox")
setDT(stress_lat_ly20_markers, keep.rownames = TRUE)[]
colnames(stress_lat_ly20_markers)[1] <- c("gene")
#stress_lat_ly20_markers <- filter(stress_lat_ly20_markers, p_val_adj <= 0.01)


neuron_npy_pos_lat_ly20_markers #25
neuron_npy_neg_lat_ly20_markers #3
satellite_lat_ly20_markers #9
schwann_lat_ly20_markers #1
macro_lat_ly20_markers #8
stress_lat_ly20_markers #1

```


```{r}

# Pre-filter features that have less than a two-fold change between the average expression
# Neurons: Find differences between Ly48 and Ly20

neuron_npy_pos_ly48_ly20_markers <- FindMarkers(object = stage_combined, 
                                               ident.1 = "NPY+ Neuron_LY48", 
                                               ident.2 = "NPY+ Neuron_LY20",
                                               verbose = FALSE, 
                                               #logfc.threshold = log(2),
                                               test.use = "wilcox")
setDT(neuron_npy_pos_ly48_ly20_markers, keep.rownames = TRUE)[]
colnames(neuron_npy_pos_ly48_ly20_markers)[1] <- c("gene")
#neuron_npy_pos_ly48_ly20_markers <- filter(neuron_npy_pos_ly48_ly20_markers, p_val_adj <= 0.01)

neuron_npy_neg_ly48_ly20_markers <- FindMarkers(object = stage_combined, 
                                               ident.1 = "NPY- Neuron_LY48", 
                                               ident.2 = "NPY- Neuron_LY20",
                                               verbose = FALSE, 
                                               #logfc.threshold = log(2), 
                                               test.use = "wilcox")
setDT(neuron_npy_neg_ly48_ly20_markers, keep.rownames = TRUE)[]
colnames(neuron_npy_neg_ly48_ly20_markers)[1] <- c("gene")
#neuron_npy_neg_ly48_ly20_markers <- filter(neuron_npy_neg_ly48_ly20_markers, p_val_adj <= 0.01)

# Satellite doesn't pass threshold
satellite_ly48_ly20_markers <- FindMarkers(object = stage_combined, 
                                          ident.1 = "Satellite_LY48", 
                                          ident.2 = "Satellite_LY20",
                                          verbose = FALSE, 
                                          #logfc.threshold = log(2), 
                                          test.use = "wilcox")
setDT(satellite_ly48_ly20_markers, keep.rownames = TRUE)[]
colnames(satellite_ly48_ly20_markers)[1] <- c("gene")
#satellite_ly48_ly20_markers <- filter(satellite_ly48_ly20_markers, p_val_adj <= 0.01)


schwann_ly48_ly20_markers <- FindMarkers(object = stage_combined, 
                                        ident.1 = "Schwann_LY48", 
                                        ident.2 = "Schwann_LY20", 
                                        verbose = FALSE, 
                                        #logfc.threshold = log(2), 
                                        test.use = "wilcox")
setDT(schwann_ly48_ly20_markers, keep.rownames = TRUE)[]
colnames(schwann_ly48_ly20_markers)[1] <- c("gene")
#schwann_ly48_ly20_markers <- filter(schwann_ly48_ly20_markers, p_val_adj <= 0.01)

macro_ly48_ly20_markers <- FindMarkers(object = stage_combined, 
                                      ident.1 = "Macrophage_LY48", 
                                      ident.2 = "Macrophage_LY20",
                                      verbose = FALSE, 
                                      #logfc.threshold = log(2), 
                                      test.use = "wilcox")
setDT(macro_ly48_ly20_markers, keep.rownames = TRUE)[]
colnames(macro_ly48_ly20_markers)[1] <- c("gene")
#macro_ly48_ly20_markers <- filter(macro_ly48_ly20_markers, p_val_adj <= 0.01)

stress_ly48_ly20_markers <- FindMarkers(object = stage_combined, 
                                       ident.1 = "Stress+ Neuron_LY48", 
                                       ident.2 = "Stress+ Neuron_LY20",
                                       verbose = FALSE, 
                                       #logfc.threshold = log(2), 
                                       test.use = "wilcox")
setDT(stress_ly48_ly20_markers, keep.rownames = TRUE)[]
colnames(stress_ly48_ly20_markers)[1] <- c("gene")
#stress_ly48_ly20_markers <- filter(stress_ly48_ly20_markers, p_val_adj <= 0.01)


neuron_npy_pos_ly48_ly20_markers #6
neuron_npy_neg_ly48_ly20_markers #1
schwann_ly48_ly20_markers #1
satellite_ly48_ly20_markers
macro_ly48_ly20_markers #5
stress_ly48_ly20_markers #0

```


---

```{r}

# Create master differential expression dataframe for each stage comparison 
# Create unfiltered dataframes:

# Uninfected & Latent
neuron_npy_pos_un_lat_markers$cell <- "NPY+ Neuron"
neuron_npy_neg_un_lat_markers$cell <- "NPY- Neuron"
schwann_un_lat_markers$cell <- "Schwann"
satellite_un_lat_markers$cell <- "Satellite"
macro_un_lat_markers$cell <- "Macrophage"
stress_un_lat_markers$cell <- "Stress+ Neuron"

un_lat <- do.call("rbind", list(neuron_npy_pos_un_lat_markers,
                                neuron_npy_neg_un_lat_markers,
                                schwann_un_lat_markers,
                                satellite_un_lat_markers,
                                macro_un_lat_markers,
                                stress_un_lat_markers))
un_lat

# Latent & LY20
neuron_npy_pos_lat_ly20_markers$cell <- "NPY+ Neuron"
neuron_npy_neg_lat_ly20_markers$cell <- "NPY- Neuron"
schwann_lat_ly20_markers$cell <- "Schwann"
satellite_lat_ly20_markers$cell <- "Satellite"
macro_lat_ly20_markers$cell <- "Macrophage"
stress_lat_ly20_markers$cell <- "Stress+ Neuron"

lat_ly20 <- do.call("rbind", list(neuron_npy_pos_lat_ly20_markers,
                                neuron_npy_neg_lat_ly20_markers,
                                schwann_lat_ly20_markers,
                                satellite_lat_ly20_markers,
                                macro_lat_ly20_markers,
                                stress_lat_ly20_markers))
lat_ly20

# LY20 & LY48
neuron_npy_pos_ly48_ly20_markers$cell <- "NPY+ Neuron"
neuron_npy_neg_ly48_ly20_markers$cell <- "NPY- Neuron"
schwann_ly48_ly20_markers$cell <- "Schwann"
satellite_ly48_ly20_markers$cell <- "Satellite"
macro_ly48_ly20_markers$cell <- "Macrophage"
stress_ly48_ly20_markers$cell <- "Stress+ Neuron"

ly48_ly20 <- do.call("rbind", list(neuron_npy_pos_ly48_ly20_markers,
                                neuron_npy_neg_ly48_ly20_markers,
                                schwann_ly48_ly20_markers,
                                satellite_ly48_ly20_markers,
                                macro_ly48_ly20_markers,
                                stress_ly48_ly20_markers))
ly48_ly20


```


---


```{r}

#### Volcano plots? with log -10(p_adjusted_value) on y axis and log2 average fold change on x axis 

# Uninfected & Latent
un_lat <- un_lat %>%
  mutate(threshold = ifelse((avg_logFC >= 0 & p_val_adj < 0.01), "up_sig",
                            ifelse((avg_logFC <= 0 & p_val_adj < 0.01) , 
                                   "down_sig", "not_sig")))
un_lat$threshold[is.na(un_lat$threshold)] <- "not_sig"

un_lat_vol <- ggplot(un_lat, aes(x = avg_logFC, y = -log10(p_val_adj))) +
  geom_point(aes(colour = threshold), size=1) +
  ggtitle("Latent vs. Uninfected") + 
  theme(plot.title = element_text(hjust = 0.5)) +
  ylim(0,50) + xlim(-2,2) +
  scale_colour_manual(values=c("blue", "grey", "red3"), 
                       name="Threshold 0.01",
                       breaks=c("down_sig", "not_sig", "up_sig"),
                       labels=c("Down", "None", "Up"))
un_lat_vol

# Latent & LY20
lat_ly20 <- lat_ly20 %>%
  mutate(threshold = ifelse((avg_logFC >= 0 & p_val_adj < 0.01), "up_sig",
                            ifelse((avg_logFC <= 0 & p_val_adj < 0.01) , 
                                   "down_sig", "not_sig")))
lat_ly20$threshold[is.na(lat_ly20$threshold)] <- "not_sig"

lat_ly20_vol <- ggplot(lat_ly20, aes(x = avg_logFC, y = -log10(p_val_adj))) +
  geom_point(aes(colour = threshold), size=1) +
  ggtitle("LY20 vs. Latent") + 
  theme(plot.title = element_text(hjust = 0.5)) +
  ylim(0,50) + xlim(-2,2) +
  scale_colour_manual(values=c("blue", "grey", "red3"), 
                       name="Threshold 0.01",
                       breaks=c("down_sig", "not_sig", "up_sig"),
                       labels=c("Down", "None", "Up"))
lat_ly20_vol

# LY20 & LY48
ly48_ly20 <- ly48_ly20 %>%
  mutate(threshold = ifelse((avg_logFC >= 0 & p_val_adj < 0.01), "up_sig",
                            ifelse((avg_logFC <= 0 & p_val_adj < 0.01) , 
                                   "down_sig", "not_sig")))
ly48_ly20$threshold[is.na(ly48_ly20$threshold)] <- "not_sig"

ly48_ly20_vol <- ggplot(ly48_ly20, aes(x = avg_logFC, y = -log10(p_val_adj))) +
  geom_point(aes(colour = threshold), size=1) +
  ggtitle("LY48 vs. LY20") + 
  theme(plot.title = element_text(hjust = 0.5)) +
  ylim(0,50) + xlim(-2,2) +
  scale_colour_manual(values=c("blue", "grey", "red3"), 
                       name="Threshold 0.01",
                       breaks=c("down_sig", "not_sig", "up_sig"),
                       labels=c("Down", "None", "Up"))
ly48_ly20_vol

```


```{r}

# Create filtered dataframe:
# Remove genes with average log fold change less than 2-fold
# Remove gene with adjusted p value greater than 0.01
un_lat <- un_lat[(un_lat$avg_logFC > log(2) | un_lat$avg_logFC < -log(2)) 
                 & un_lat$p_val_adj < 0.01]

lat_ly20 <- lat_ly20[(lat_ly20$avg_logFC > log(2) | lat_ly20$avg_logFC < -log(2)) 
                     & lat_ly20$p_val_adj < 0.01]

ly48_ly20 <- ly48_ly20[(ly48_ly20$avg_logFC > log(2) | ly48_ly20$avg_logFC < -log(2))
                       & ly48_ly20$p_val_adj < 0.01]

# This is what I will run the pathway analysis on 
un_lat
lat_ly20
ly48_ly20

# Write out dataframes
write.table(un_lat, file = "uninfected_latent_sig_genes.txt", sep = "\t", quote = FALSE)
write.table(lat_ly20, file = "latent_ly20_sig_genes.txt", sep = "\t", quote = FALSE)
write.table(ly48_ly20, file = "ly20_ly48_sig_genes.txt", sep = "\t", quote = FALSE)

```


---



```{r}

# What proportion of cells are in each cluster? 

# Get Frequency
stage_freq <- as.data.frame(table(Idents(stage_combined)))


# Get Stage total count (manually add)
stage_freq$count <- c(rep(3435, 6), rep(2127, 6), rep(7850, 6), rep(5552, 6))

# Get proportion
stage_freq$prop <- (stage_freq$Freq/stage_freq$count)
stage_freq

# Split cell type and stage into two columns
stage_freq <- separate(stage_freq, Var1, into=c("cell_type", "stage"), sep="_" )

# Order by cell types
stage_freq <- stage_freq[order(stage_freq$cell_type), ]

# Write out frequency table
write.table(stage_freq, file = "stage_cell_frequency_prop.txt", sep = "\t", quote = FALSE)

# View
stage_freq

# Plot
stage_freq_plot <- ggplot(data = stage_freq, 
                          mapping = aes(x = stage, y = prop, col = cell_type, group = cell_type)) +
  geom_point() + geom_line() + 
  scale_x_discrete(limits=c("Uninfected","Latent","LY20", "LY48")) +
  theme_grey()

stage_freq_plot

```


---




---





---

Extra Notes:


---

find markers results:

The results data frame has the following columns :
p_val : p_val (unadjusted)
avg_logFC : log fold-chage of the average expression between the two groups. Positive values indicate that the gene is more highly expressed in the first group.
pct.1 : The percentage of cells where the gene is detected in the first group
pct.2 : The percentage of cells where the gene is detected in the second group
p_val_adj : Adjusted p-value, based on bonferroni correction using all genes in the dataset.

---

# Also to subset by gene...
https://github.com/satijalab/seurat/issues/349


---



---

---



---