---
title: "R Notebook"
output: html_notebook
---


---

Shaleigh Smith
NGS Final

---

Cellranger Results:
- Unfiltered: Contains every barcode from fixed list of known-good barcode sequences. This includes background and cell associated barcodes.
- Filtered: Contains only detected cellular barcodes.

---

General Steps:
1) Initialize a Seurat object
2) Filter cells based on built-in metrics
3) Normalize the data
4) Select variable genes across cells
5) Removing confounding factors by regression
6) Linear dimensionality reduction
7) PCA-based clustering
8) t-SNE visualization 
9) Identify differentially expressed genes

---

# Helpful Links:

https://github.com/igordot/seurat/blob/master/R/preprocessing.R
https://satijalab.org/seurat/v3.0/immune_alignment.html 
https://github.com/satijalab/seurat/wiki 
https://broadinstitute.github.io/2019_scWorkshop/index.html 
https://hemberg-lab.github.io/scRNA.seq.course/index.html
#https://satijalab.org/seurat/v3.0/pbmc3k_tutorial.html


---


```{r}

# Load libraries
library(tidyverse)
library(dplyr)
library(Matrix)
library(devtools)
library(Seurat)
library(ggplot2)
library(circlize)

packageVersion("Seurat")

```


---


Preprocessing

---


```{r}

# Set working directory
setwd("/Users/sha/Desktop/NGS_Informatics/NGS_final_project/")

### Preprocessing 
# Import matrices 
# Create Seurat objects 
# Set minimum for cells (3) and features (200) to remove low quality cells 

# Uninfected
data_dir_un <- "./uninfected/filtered_feature_bc_matrix/"
list.files(data_dir_un) # Should show barcodes.tsv, genes.tsv, and matrix.mtx
exp_matrix_un <- Read10X(data.dir = data_dir_un)
un <- CreateSeuratObject(counts = exp_matrix_un, project = "uninfected",
                         min.cells = 3, min.features = 200)
un$comp <- "uninfected"
un


# Latent
data_dir_lat <- "./latent/filtered_feature_bc_matrix/"
list.files(data_dir_lat) # Should show barcodes.tsv, genes.tsv, and matrix.mtx
exp_matrix_lat <- Read10X(data.dir = data_dir_lat)
lat = CreateSeuratObject(counts = exp_matrix_lat, project = "latent",
                         min.cells = 3, min.features = 200)
lat$comp <- "latent"
lat


# Ly20
data_dir_ly20 <- "./ly20/filtered_feature_bc_matrix/"
list.files(data_dir_ly20) # Should show barcodes.tsv, genes.tsv, and matrix.mtx
exp_matrix_ly20 <- Read10X(data.dir = data_dir_ly20)
ly20 = CreateSeuratObject(counts = exp_matrix_ly20, project = "ly20",
                         min.cells = 3, min.features = 200)
ly20$comp <- "ly20"
ly20


# Uninfected
data_dir_ly48 <- "./ly48/filtered_feature_bc_matrix/"
list.files(data_dir_ly48) # Should show barcodes.tsv, genes.tsv, and matrix.mtx
exp_matrix_ly48 <- Read10X(data.dir = data_dir_ly48)
ly48 = CreateSeuratObject(counts = exp_matrix_ly48, project = "ly48",
                         min.cells = 3, min.features = 200)
ly48$comp <- "ly48"
ly48 

```

```{r}

# View examples
colnames(un)
rownames(un)
head(x = un@meta.data, 20)
un[["RNA"]]@data

?seurat

```


```{r}

### Quality Control & Filtering

# View summary of total expression by single cell
summary(colSums(un))
summary(colSums(lat))
summary(colSums(ly20))
summary(colSums(ly48))

### Note that the mean decreases from uninfected to latent to ly 
### This suggests that host gene expression is decreasing as viral expression increases 


# Calculate percentage of all counts belonging to a subset of the possible features for each cell
VlnPlot(object = un, features = c("nFeature_RNA", "nCount_RNA"), ncol = 2, 
        cols = c("#007BC3"),  pt.size = 0.25)
VlnPlot(object = lat, features = c("nFeature_RNA", "nCount_RNA"), ncol = 2, 
        cols = c("#007BC3"),  pt.size = 0.25)
VlnPlot(object = ly20, features = c("nFeature_RNA", "nCount_RNA"), ncol = 2, 
        cols = c("#007BC3"),  pt.size = 0.25)
VlnPlot(object = ly48, features = c("nFeature_RNA", "nCount_RNA"), ncol = 2, 
        cols = c("#007BC3"),  pt.size = 0.25)

# Visualize feature relationships: Count_RNA vs. Feature_RNA
scatter_un_1 <- FeatureScatter(object = un, feature1 = "nCount_RNA", feature2 = "nFeature_RNA",
                               col = "darkblue") + xlim(c(0, 85000)) + ylim(c(0, 8000))
scatter_lat_1 <- FeatureScatter(object = lat, feature1 = "nCount_RNA", feature2 = "nFeature_RNA",
                                col = "darkred") + xlim(c(0, 85000)) + ylim(c(0, 8000))
scatter_ly20_1 <- FeatureScatter(object = ly20, feature1 = "nCount_RNA", feature2 = "nFeature_RNA",
                                 col = "darkgoldenrod2") + xlim(c(0, 85000)) + ylim(c(0, 8000))
scatter_ly48_1 <- FeatureScatter(object = ly48, feature1 = "nCount_RNA", feature2 = "nFeature_RNA",
                                 col = "darkgreen") + xlim(c(0, 85000)) + ylim(c(0, 8000))
scatter_un_1
scatter_lat_1
scatter_ly20_1
scatter_ly48_1
#CombinePlots(plots = list(scatter_un_1, scatter_lat_1, scatter_ly20_1, scatter_ly48_1), ncol = 2)

### All appear to be a horizontal parabola (k * sqrt(x) relationship)
### All have a fairly high positive correlation (greater than 0.87 )

```


```{r}

# Distribution of expression levels by Feature and Cell
hist(rowSums(un),
     breaks = 100, main = "Uninfected: Expression sum by Feature",
     xlab = "Sum Expression")
hist(colSums(un),
     breaks = 100, main = "Uninfected: Expression sum by Cell",
     xlab = "Sum Expression")

hist(rowSums(lat),
     breaks = 100, main = "Latent: Expression sum by Feature",
     xlab = "Sum Expression")
hist(colSums(lat),
     breaks = 100, main = "Latent: Expression sum by Cell",
     xlab = "Sum Expression")

hist(rowSums(ly20),
     breaks = 100, main = "LY20: Expression sum by Feature",
     xlab = "Sum Expression")
hist(colSums(ly20),
     breaks = 100, main = "LY20: Expression sum by Cell",
     xlab = "Sum Expression")

hist(rowSums(ly48),
     breaks = 100, main = "LY48: Expression sum by Feature",
     xlab = "Sum Expression")
hist(colSums(ly48),
     breaks = 100, main = "LY48: Expression sum by Cell",
     xlab = "Sum Expression")

```



```{r}

### Normalize each gene expression matrix 
# Do Log Normalization first with default scale

# Uninfected Log Normalization
un_norm_log <- NormalizeData(un, assay = NULL, normalization.method = "LogNormalize", 
                         scale.factor = 10000)

hist(rowSums(un_norm_log),
     breaks = 100, main = "Uninfected Log Normalized: Expression sum by Feature",
     xlab = "Sum Expression")
hist(colSums(un_norm_log),
     breaks = 100, main = "Uninfected Log Normalized: Expression sum by Cell",
     xlab = "Sum Expression")

# Latent Log Normalization
lat_norm_log <- NormalizeData(lat, assay = NULL, normalization.method = "LogNormalize", 
                         scale.factor = 10000)

hist(rowSums(lat_norm_log),
     breaks = 100, main = "Latent Log Normalized: Expression sum by Feature",
     xlab = "Sum Expression")
hist(colSums(lat_norm_log),
     breaks = 100, main = "Latent Log Normalized: Expression sum by Cell",
     xlab = "Sum Expression")

# ly20 Log Normalization
ly20_norm_log <- NormalizeData(ly20, assay = NULL, normalization.method = "LogNormalize", 
                         scale.factor = 10000)

hist(rowSums(ly20_norm_log),
     breaks = 100, main = "LY20 Log Normalized: Expression sum by Feature",
     xlab = "Sum Expression")
hist(colSums(ly20_norm_log),
     breaks = 100, main = "LY20 Log Normalized: Expression sum by Cell",
     xlab = "Sum Expression")

# ly48 Log Normalization
ly48_norm_log <- NormalizeData(ly48, assay = NULL, normalization.method = "LogNormalize", 
                         scale.factor = 10000)

hist(rowSums(ly48_norm_log),
     breaks = 100, main = "LY48 Log Normalized: Expression sum by Feature",
     xlab = "Sum Expression")
hist(colSums(ly48_norm_log),
     breaks = 100, main = "LY48 Log Normalized: Expression sum by Cell",
     xlab = "Sum Expression")

```

```{r}

### Check CLR Normalization for comparison

# Uninfected CLR Normalization
#un_norm_clr <- NormalizeData(un, assay = NULL, normalization.method = "CLR", 
#                         scale.factor = 10000, margin = 1)
#hist(rowSums(un_norm_clr),
#     breaks = 100, main = "Uninfected CLR Normalized: Expression sum by Feature",
#     xlab = "Sum Expression")
#hist(colSums(un_norm_clr),
#     breaks = 100, main = "Uninfected CLR Normalized: Expression sum by Cell",
#     xlab = "Sum Expression")

# Latent CLR Normalization
#lat_norm_clr <- NormalizeData(lat, assay = NULL, normalization.method = "CLR", 
#                         scale.factor = 10000, margin = 1)
#hist(rowSums(lat_norm_clr),
#     breaks = 100, main = "Latent CLR Normalized: Expression sum by Feature",
#     xlab = "Sum Expression")
#hist(colSums(lat_norm_clr),
#     breaks = 100, main = "Latent CLR Normalized: Expression sum by Cell",
#     xlab = "Sum Expression")

# ly20 CLR Normalization
#ly20_norm_clr <- NormalizeData(ly20, assay = NULL, normalization.method = "CLR", 
#                         scale.factor = 10000, margin = 1)
#hist(rowSums(ly20_norm_clr),
#     breaks = 100, main = "LY20 CLR Normalized: Expression sum by Feature",
#     xlab = "Sum Expression")
#hist(colSums(ly20_norm_clr),
#     breaks = 100, main = "LY20 CLR Normalized: Expression sum by Cell",
#     xlab = "Sum Expression")

# ly48 CLR Normalization
#ly48_norm_clr <- NormalizeData(ly48, assay = NULL, normalization.method = "CLR", 
#                         scale.factor = 10000, margin = 1)
#hist(rowSums(ly48_norm_clr),
#     breaks = 100, main = "LY48 CLR Normalized: Expression sum by Feature",
#     xlab = "Sum Expression")
#hist(colSums(ly48_norm_clr),
#     breaks = 100, main = "LY48 CLR Normalized: Expression sum by Cell",
#     xlab = "Sum Expression")


# This doesn't work as well 

```


```{r}

### Remove unwanted cells from the dataset to decrease noise after normalization

# The normalized uninfected appears to be bimodal
# To decrease the smalled leftward peak, increase the minimum number of features by cell

un_sub <- subset(x = un, subset = nFeature_RNA > 300 & nFeature_RNA < 6000)
lat_sub <- subset(x = lat, subset = nFeature_RNA > 300 & nFeature_RNA < 6000)
ly20_sub <- subset(x = ly20, subset = nFeature_RNA > 300 & nFeature_RNA < 6000)
ly48_sub <- subset(x = ly48, subset = nFeature_RNA > 300 & nFeature_RNA < 6000)


```

```{r}

### Do normalization on subsetted matrix
# Use Log Normalization
# Alter subset to increase normal distribution

### Normalize each gene expression matrix 
# Do Log Normalization first with default scale

# Uninfected Log Normalization
un_sub_norm_log <- NormalizeData(un_sub, assay = NULL, normalization.method = "LogNormalize", 
                         scale.factor = 10000)

hist(rowSums(un_sub_norm_log),
     breaks = 100, main = "Uninfected Sub Log Normalized: Expression sum by Feature",
     xlab = "Sum Expression")
hist(colSums(un_sub_norm_log),
     breaks = 100, main = "Uninfected Sub Log Normalized: Expression sum by Cell",
     xlab = "Sum Expression")

# Latent Log Normalization
lat_sub_norm_log <- NormalizeData(lat_sub, assay = NULL, normalization.method = "LogNormalize", 
                         scale.factor = 10000)

hist(rowSums(lat_sub_norm_log),
     breaks = 100, main = "Latent Sub Log Normalized: Expression sum by Feature",
     xlab = "Sum Expression")
hist(colSums(lat_sub_norm_log),
     breaks = 100, main = "Latent Sub Log Normalized: Expression sum by Cell",
     xlab = "Sum Expression")

# ly20 Log Normalization
ly20_sub_norm_log <- NormalizeData(ly20_sub, assay = NULL, normalization.method = "LogNormalize", 
                         scale.factor = 10000)

hist(rowSums(ly20_sub_norm_log),
     breaks = 100, main = "LY20 Sub Log Normalized: Expression sum by Feature",
     xlab = "Sum Expression")
hist(colSums(ly20_sub_norm_log),
     breaks = 100, main = "LY20 Sub Log Normalized: Expression sum by Cell",
     xlab = "Sum Expression")

# ly48 Log Normalization
ly48_sub_norm_log <- NormalizeData(ly48_sub, assay = NULL, normalization.method = "LogNormalize", 
                         scale.factor = 10000)

hist(rowSums(ly48_sub_norm_log),
     breaks = 100, main = "LY48 Sub Log Normalized: Expression sum by Feature",
     xlab = "Sum Expression")
hist(colSums(ly48_sub_norm_log),
     breaks = 100, main = "LY48 Sub Log Normalized: Expression sum by Cell",
     xlab = "Sum Expression")

```

---

```{r}

### Feature Selection
# Identify the most highly variable features

un_feat <- FindVariableFeatures(object = un_sub_norm_log, selection.method = "vst", nfeatures = 2000)
lat_feat <- FindVariableFeatures(object = lat_sub_norm_log, selection.method = "vst", nfeatures = 2000)
ly20_feat <- FindVariableFeatures(object = ly20_sub_norm_log, selection.method = "vst", nfeatures = 2000)
ly48_feat <- FindVariableFeatures(object = ly48_sub_norm_log, selection.method = "vst", nfeatures = 2000)


# Identify the 10 most highly variable genes
un_top10 <- head(x = VariableFeatures(object = un_feat), 10)
lat_top10 <- head(x = VariableFeatures(object = lat_feat), 10)
ly20_top10 <- head(x = VariableFeatures(object = ly20_feat), 10)
ly48_top10 <- head(x = VariableFeatures(object = ly48_feat), 10)

# Plot variable features with labels
un_feat_plot <- LabelPoints(plot = VariableFeaturePlot(object = un_feat, 
                                                       cols = c("black", "darkblue")), 
                            points = un_top10, repel = TRUE)
lat_feat_plot <- LabelPoints(plot = VariableFeaturePlot(object = lat_feat, 
                                                        cols = c("black", "darkred")), 
                            points = lat_top10, repel = TRUE)
ly20_feat_plot <- LabelPoints(plot = VariableFeaturePlot(object = ly20_feat, 
                                                         cols = c("black", "darkgoldenrod2")), 
                            points = ly20_top10, repel = TRUE)
ly48_feat_plot <- LabelPoints(plot = VariableFeaturePlot(object = ly48_feat, 
                                                         cols = c("black", "darkgreen")), 
                            points = ly48_top10, repel = TRUE)

un_feat_plot
lat_feat_plot
ly20_feat_plot
ly48_feat_plot

```


```{r}

### Scale Data
# Shifts the expression of each gene, so that the mean expression across cells is 0
# Scales the expression of each gene, so that the variance across cells is 1
# This step gives equal weight in downstream analyses, so that highly-expressed genes do not dominate

un_genes <- rownames(x = un_feat)
un_scale <- ScaleData(object = un_feat, features = un_genes)

lat_genes <- rownames(x = lat_feat)
lat_scale <- ScaleData(object = lat_feat, features = lat_genes)

ly20_genes <- rownames(x = ly20_feat)
ly20_scale <- ScaleData(object = ly20_feat, features = ly20_genes)

ly48_genes <- rownames(x = ly48_feat)
ly48_scale <- ScaleData(object = ly48_feat, features = ly48_genes)

```


---

```{r}

### Linear Dimensional Reduction
# PCA

un_scale <- RunPCA(object = un_scale, features = VariableFeatures(object = un_scale))
lat_scale <- RunPCA(object = lat_scale, features = VariableFeatures(object = lat_scale))
ly20_scale <- RunPCA(object = ly20_scale, features = VariableFeatures(object = ly20_scale))
ly48_scale <- RunPCA(object = ly48_scale, features = VariableFeatures(object = ly48_scale))

# View PCA features
print(x = un_scale[["pca"]], dims = 1:5, nfeatures = 5)
print(x = lat_scale[["pca"]], dims = 1:5, nfeatures = 5)
print(x = ly20_scale[["pca"]], dims = 1:5, nfeatures = 5)
print(x = ly48_scale[["pca"]], dims = 1:5, nfeatures = 5)

# Get and set variable feature information
un_feat_info <- VizDimLoadings(object = un_scale, dims = 1:2, reduction = "pca")
un_feat_info
lat_feat_info <- VizDimLoadings(object = lat_scale, dims = 1:2, reduction = "pca")
ly20_feat_info <- VizDimLoadings(object = ly20_scale, dims = 1:2, reduction = "pca")
ly48_feat_info <- VizDimLoadings(object = ly48_scale, dims = 1:2, reduction = "pca")

# View Dimensional Reduction Plot 
un_pca <- DimPlot(object = un_scale, reduction = "pca", cols = c("darkblue", "darkblue"))
un_pca
lat_pca <- DimPlot(object = lat_scale, reduction = "pca", cols = c("darkred", "darkred"))
lat_pca
ly20_pca <- DimPlot(object = ly20_scale, reduction = "pca", cols = c("darkgoldenrod2", "darkgoldenrod2"))
ly20_pca
ly48_pca <- DimPlot(object = ly48_scale, reduction = "pca", cols = c("darkgreen", "darkgreen"))
ly48_pca



##################################
# View Heatmap

###### FIX colors

#ramp <- colorRamp2(c(-2, 0, 2), c("blue", "white", "red"))
#un_gg <- DimHeatmap(object = un_scale, dims = 1:2, cells = 500, fast = FALSE)
#class(un_gg)
#final <- un_gg + geom_raster() + scale_fill_gradientn(colours=c("#0000FFFF","#FFFFFFFF","#FF0000FF"))
#final

DimHeatmap(object = un_scale, dims = 1:2, cells = 500, balanced = TRUE, fast = FALSE)
DimHeatmap(object = lat_scale, dims = 1:2, cells = 500, balanced = TRUE, fast = FALSE)
DimHeatmap(object = ly20_scale, dims = 1:2, cells = 500, balanced = TRUE, fast = FALSE)
DimHeatmap(object = ly48_scale, dims = 1:2, cells = 500, balanced = TRUE, fast = FALSE)


```

```{r}

### Determining the Dimensionality of the dataset
# How many PCs are optimal 

# Elbow Plot
ElbowPlot(object = un_scale) # 14-15 PCs
ElbowPlot(object = lat_scale) # 12-13 PCs
ElbowPlot(object = ly20_scale) # 5-6 PCs
ElbowPlot(object = ly48_scale) # 7-6 PCs

# Compare the distribution of p-values for each PC with a uniform distribution (dashed line)
un_scale <- JackStraw(object = un_scale, num.replicate = 100)
un_scale <- ScoreJackStraw(object = un_scale, dims = 1:20)
JackStrawPlot(object = un_scale, dims = 1:20)

lat_scale <- JackStraw(object = lat_scale, num.replicate = 100)
lat_scale <- ScoreJackStraw(object = lat_scale, dims = 1:20)
JackStrawPlot(object = lat_scale, dims = 1:20)

ly20_scale <- JackStraw(object = ly20_scale, num.replicate = 100)
ly20_scale <- ScoreJackStraw(object = ly20_scale, dims = 1:20)
JackStrawPlot(object = ly20_scale, dims = 1:20)

ly48_scale <- JackStraw(object = ly48_scale, num.replicate = 100)
ly48_scale <- ScoreJackStraw(object = ly48_scale, dims = 1:20)
JackStrawPlot(object = ly48_scale, dims = 1:20)

```

```{r}

# Cluster Cells

# Uninfected
un_scale <- FindNeighbors(object = un_scale, dims = 1:20)
un_scale <- FindClusters(object = un_scale, resolution = 0.5)
head(x = Idents(object = un_scale), 10)

# Latent
lat_scale <- FindNeighbors(object = lat_scale, dims = 1:20)
lat_scale <- FindClusters(object = lat_scale, resolution = 0.5)
head(x = Idents(object = lat_scale), 10)

# LY20
ly20_scale <- FindNeighbors(object = ly20_scale, dims = 1:20)
ly20_scale <- FindClusters(object = ly20_scale, resolution = 0.5)
head(x = Idents(object = ly20_scale), 10)

# LY48
ly48_scale <- FindNeighbors(object = ly48_scale, dims = 1:20)
ly48_scale <- FindClusters(object = ly48_scale, resolution = 0.5)
head(x = Idents(object = ly48_scale), 10)

```

```{r}

set.seed(8888)
# TSNE 

un_scale <- RunTSNE(object = un_scale, dims = 1:20)
un_tsne <- DimPlot(object = un_scale, reduction = "tsne")
un_tsne

lat_scale <- RunTSNE(object = lat_scale, dims = 1:20)
lat_tsne <- DimPlot(object = lat_scale, reduction = "tsne")
lat_tsne

ly20_scale <- RunTSNE(object = ly20_scale, dims = 1:20)
ly20_tsne <- DimPlot(object = ly20_scale, reduction = "tsne")
ly20_tsne

ly48_scale <- RunTSNE(object = ly48_scale, dims = 1:20)
ly48_tsne <- DimPlot(object = ly48_scale, reduction = "tsne")
ly48_tsne

```

```{r}

set.seed(8888)
# UMAP

un_scale <- RunUMAP(object = un_scale, dims = 1:20)
un_umap <- DimPlot(object = un_scale, reduction = "umap")
un_umap

lat_scale <- RunUMAP(object = lat_scale, dims = 1:20)
lat_umap <- DimPlot(object = lat_scale, reduction = "umap")
lat_umap

ly20_scale <- RunUMAP(object = ly20_scale, dims = 1:20)
ly20_umap <- DimPlot(object = ly20_scale, reduction = "umap")
ly20_umap

ly48_scale <- RunUMAP(object = ly48_scale, dims = 1:20)
ly48_umap <- DimPlot(object = ly48_scale, reduction = "umap")
ly48_umap

```


```{r}

# Save objects
saveRDS(un_scale, file = "./uninfected_scale.rds")
saveRDS(lat_scale, file = "./latent_scale.rds")
saveRDS(ly20_scale, file = "./ly20_scale.rds")
saveRDS(ly48_scale, file = "./ly48_scale.rds")

```


---

```{r}


# Export Figures

# Normalized Distribution
png("uninfected_sub_norm_log_hist.png", units="in", width=8, height=5, res=600) 
hist(colSums(un_sub_norm_log),
     breaks = 100, main = "Uninfected Sub Log Normalized: Expression sum by Cell",
     xlab = "Sum Expression",
     col = "darkblue")
dev.off()

png("latent_sub_norm_log_hist.png", units="in", width=8, height=5, res=600) 
hist(colSums(lat_sub_norm_log),
     breaks = 100, main = "Latent Sub Log Normalized: Expression sum by Cell",
     xlab = "Sum Expression",
     col = "darkred")
dev.off()

png("ly20_sub_norm_log_hist.png", units="in", width=8, height=5, res=600) 
hist(colSums(ly20_sub_norm_log),
     breaks = 100, main = "LY20 Sub Log Normalized: Expression sum by Cell",
     xlab = "Sum Expression",
     col = "darkgoldenrod2")
dev.off()

png("ly48_sub_norm_log_hist.png", units="in", width=8, height=5, res=600) 
hist(colSums(ly48_sub_norm_log),
     breaks = 100, main = "LY48 Sub Log Normalized: Expression sum by Cell",
     xlab = "Sum Expression",
     col = "darkgreen")
dev.off()

# Labeled top 10 feature plots
png("uninfected_feature_plot_top_10.png", units="in", width=8, height=5, res=600) 
un_feat_plot
dev.off()

png("latent_feature_plot_top_10.png", units="in", width=8, height=5, res=600) 
lat_feat_plot
dev.off()

png("ly20_feature_plot_top_10.png", units="in", width=8, height=5, res=600) 
ly20_feat_plot
dev.off()

png("ly48_feature_plot_top_10.png", units="in", width=8, height=5, res=600) 
ly48_feat_plot
dev.off()

# Feature info
png("uninfected_feature_info.png", units="in", width=8, height=5, res=600) 
un_feat_info
dev.off()

png("latent_feature_info.png", units="in", width=8, height=5, res=600) 
lat_feat_info
dev.off()

png("ly20_feature_info.png", units="in", width=8, height=5, res=600) 
ly20_feat_info
dev.off()

png("ly48_feature_info.png", units="in", width=8, height=5, res=600) 
ly48_feat_info
dev.off()


# PCA
png("uninfected_pca.png", units="in", width=8, height=5, res=600) 
un_pca
dev.off()

png("latent_pca.png", units="in", width=8, height=5, res=600) 
lat_pca
dev.off()

png("ly20_pca.png", units="in", width=8, height=5, res=600) 
ly20_pca
dev.off()

png("ly48_pca.png", units="in", width=8, height=5, res=600) 
ly48_pca
dev.off()

# Elbow Plots
png("uninfected_elbow.png", units="in", width=8, height=5, res=600) 
ElbowPlot(object = un_scale)
dev.off()

png("latent_elbow.png", units="in", width=8, height=5, res=600) 
ElbowPlot(object = lat_scale)
dev.off()

png("ly20_elbow.png", units="in", width=8, height=5, res=600) 
ElbowPlot(object = ly20_scale)
dev.off()

png("ly48_elbow.png", units="in", width=8, height=5, res=600) 
ElbowPlot(object = ly48_scale)
dev.off()


# PCA p-value distribution (jackstraw)
png("uninfected_jackstraw.png", units="in", width=8, height=5, res=600) 
JackStrawPlot(object = un_scale, dims = 1:20)
dev.off()

png("latent_jackstraw.png", units="in", width=8, height=5, res=600) 
JackStrawPlot(object = lat_scale, dims = 1:20)
dev.off()

png("ly20_jackstraw.png", units="in", width=8, height=5, res=600) 
JackStrawPlot(object = ly20_scale, dims = 1:20)
dev.off()

png("ly48_jackstraw.png", units="in", width=8, height=5, res=600) 
JackStrawPlot(object = ly48_scale, dims = 1:20)
dev.off()


# TSNE
png("uninfected_tsne.png", units="in", width=8, height=5, res=600) 
un_tsne
dev.off()

png("latent_tsne.png", units="in", width=8, height=5, res=600) 
lat_tsne
dev.off()

png("ly20_tsne.png", units="in", width=8, height=5, res=600) 
ly20_tsne
dev.off()

png("ly48_tsne.png", units="in", width=8, height=5, res=600) 
ly48_tsne
dev.off()


# UMAP
png("uninfected_umap.png", units="in", width=8, height=5, res=600) 
un_umap
dev.off()

png("latent_umap.png", units="in", width=8, height=5, res=600) 
lat_umap
dev.off()

png("ly20_umap.png", units="in", width=8, height=5, res=600) 
ly20_umap
dev.off()

png("ly48_umap.png", units="in", width=8, height=5, res=600) 
ly48_umap
dev.off()

```


---


```{r}

# Integrate data for comparisons (uninfected vs latent, latent vs ly20, ly20 vs ly48)

# Ex.
#immune.anchors <- FindIntegrationAnchors(object.list = list(ctrl, stim), dims = 1:20)
#immune.combined <- IntegrateData(anchorset = immune.anchors, dims = 1:20)

# https://satijalab.org/seurat/v3.0/immune_alignment.html 

```


---


---